app/controllers/application_controller.rb

```diff
@@ -29,6 +29,12 @@ def read_only_error
     service_unavailable(:error => 'Sorry, actions that write to the database are currently disabled for maintenance.')
   end
 
+  def merge_embed_config(presenter)
```

Comment:
> I was a little unclear on what merge_embed_config was supposed to do. would filter_embed_keys make more sense?

What it does requires enough knowledge that no method name will be perfect, but it can definitely be improved. It:

1. Starts with request `params` and a presenter class (like `DC::Embed::Document`)
2. Filters the params by the presenter's valid config keys
3. Creates a new presenter Config so we can run `dump`, which coerces the types and converts keys to the `output_as` versions (if the DOM mechanism calls for it)

So the inputs are the presenter class, the request `params`, and the existing `@embed_options` instance var hash. It modifies `@embed_options`, and its return value doesn't matter.

Make sense? Any better method names?

> Additionally, this probably isn't generic functionality that should be in application_controller but in a mixin/concern/something that controllers that handle embeddable resources opt into.

True enough. I had this in each of the relevant controllers (document/page/annotation) and was trying to DRY it up a bit, but this was probably the wrong way. I'm a bit out of my depth here; should I just move it back to each controller with a TODO for DRYing it up?